### ISSUE TITLE
patchGAN

### ISSUE THREAD

**Issue by octaveguinebretiere** [Changchun-Yang/LoftGAN#2]
Hello,

I was very interested to read about your paper "Deep Learning Enables Robust and Precise Light Focusing on Treatment Needs" . However, I can not find anything related to the patchGAN in your code, did you implement it somewhere? I would be interested.

Thank you in advance. 

Octave

**Comment by Changchun-Yang** [comment]
Hi,
PatchGAN is special mainly in the discriminator, which outputs an N*N matrix, different from the traditional GAN's discriminator which outputs a true or false vector. So we only used the patchGAN design on the discriminator. Hope my answer is helpful to you.

Changchun

**Comment by octaveguinebretiere** [comment]
Hi, 

Thanks for your answer, it's clear how you implement the patchGAN design on the discriminator, however, I can not find where you implement the style loss, could you provide me insight on that?

Thank you in advance.

Kinds,

Octave

**Comment by Changchun-Yang** [comment]
Hi,
In fact, style loss is not implemented explicitly. The style loss in the paper is about the expectation of different item: the output of last layer about y, y_tilde, y_hat. So we define: "discriminator_train = Model([x, z_p], [dis_x, dis_x_tilde, dis_x_p], name='di')", then we use "discriminator_train.compile(rmsprop, ['binary_crossentropy'] * 3, ['acc'] * 3)", which "['binary_crossentropy'] * 3" will automatically optimize the joint expectation about those three. This is achieved based on the characteristics of keras. If you use pytorch, it will be easier to implement and easier to read.

Regards,
Changchun
